File: practicas/practica_2/deepspace_Java_P2/src/main/java/deepspace/HangarToUI.java
/**
 * @file HangarToUI.java
 * @author Profesor
 */
package deepspace;

import java.util.ArrayList;

/**
 * @brief Representación ToUI de Hangar
 */
public class HangarToUI {
    private int maxElements;
    private ArrayList<WeaponToUI> weapons;
    private ArrayList<ShieldToUI> shieldBoosters;    
    
    /**
     * @brief Constructor con un parámetro
     * @param h Instancia de la clase Hangar
     */
    HangarToUI(Hangar h) {
        weapons=new ArrayList<>();
        shieldBoosters=new ArrayList<>(); 
    
        maxElements=h.getMaxElements();
        
        for (Weapon w:h.getWeapons()) {
            weapons.add(w.getUIversion());
        }
        
        for(ShieldBooster s:h.getShieldBoosters()) {
            shieldBoosters.add(s.getUIversion());
        }
    }
    
    /**
     * @brief Consultor de la capacidad máxima de elementos dentro del hangar
     * @return maxElements
     */
    public int getMaxElements() {
        return maxElements;
    }

    /**
     * @brief Consultor de la colección de armas concretas disponibles 
     *        dentro del hangar
     * @return weapons Array con las armas del hangar
     */
    public ArrayList<WeaponToUI> getWeapons() {
        return weapons;
    }

    /**
     * @brief Consultor de los potenciadores de escudo disponibles dentro del
     *        hangar
     * @return shieldBoosters Array con los potenciadores de escudo del hangar
     */
    public ArrayList<ShieldToUI> getShieldBoosters() {
        return shieldBoosters;
    }
}

File: practicas/practica_2/Deepspace_Ruby_P2/lib/HangarToUI.rb
#encoding:utf-8

# File: HangarToUI.rb
# Author: Profesor

module Deepspace

# Brief: Representación ToUI de Hangar
class HangarToUI 

  # Consultores
  attr_reader :maxElements, :weapons, :shieldBoosters
  
  # Brief: Constructor con un parámetro
  # Param h: Instancia de la clase Hangar
  def initialize(h) 
    @weapons=Array.new()
    @shieldBoosters=Array.new()
    @maxElements=h.maxElements
    
    for w in h.weapons do
      @weapons.push(w.getUIversion())
    end
    
    for s in h.shieldBoosters do
      @shieldBoosters.push(s.getUIversion())
    end
  end
  
  # Brief: Función para representar una instancia de la clase HangarToUI
  #        en un string
  # Return: out String que representa una instancia de la clase HangarToUI
  def to_s
    return "Max Elements: #{@maxElements}, Weapons: [#{@weapons.join(", ")}], Shields: [#{@shieldBoosters.join(", ")}]"
  end

end # Class

end # Module